pkg/dialer/dialer.go

```diff
@@ -28,8 +29,19 @@ type dialResult struct {
 	err error
 }
 
+// ContextDialer returns a GRPC net.Conn connected to the provided address
+func ContextDialer(ctx context.Context, address string) (net.Conn, error) {
```

Comment:
can we support ctx.Cancel?

Reply:
My current approach is a bit dirty yeah; I'm not too familiar with these, so better suggestions are definitely welcome (was also thinking of reversing the logic and use context as the default)

If you can show a quick example, I'm happy to add it 😅